test/Pomelo.EntityFrameworkCore.MySql.PerfTests/Models/Crm.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
+
+namespace Pomelo.EntityFrameworkCore.MySql.PerfTests.Models
+{
+
+	public static class CrmMeta
+	{
+		public static void OnModelCreating(ModelBuilder modelBuilder)
+		{
+			modelBuilder.Entity<CrmAdminRole>(entity =>
+			{
+				entity.HasKey(am => new {am.AdminId, am.RoleId});
+
+				entity.HasOne(am => am.Admin)
+					.WithMany(a => a.AdminRoles)
+					.HasForeignKey(am => am.AdminId);
+
+				entity.HasOne(am => am.Role)
+					.WithMany(m => m.AdminRoles)
+					.HasForeignKey(am => am.RoleId);
+			});
+
+			modelBuilder.Entity<CrmAdminMenu>(entity =>
+			{
+				entity.HasKey(am => new {am.AdminId, am.MenuId});
+
+				entity.HasOne(am => am.Admin)
+					.WithMany(a => a.AdminMenus)
+					.HasForeignKey(am => am.AdminId);
+
+				entity.HasOne(am => am.Menu)
+					.WithMany(m => m.AdminMenus)
+					.HasForeignKey(am => am.MenuId);
+			});
+
+		}
+	}
+
+	public class CrmAdmin
+	{
+		public int Id { get; set; }
+
+		[MaxLength(10)]
```

Comment:
All things looks great but please keep the consistency, if you are using Fluent API, please change `[MaxLength(10)]` into `entity.Property(x => x.Username).HasMaxLength(10)`, or use data annotations instead of all Fluent API.